### ISSUE TITLE
Workers get stuck at drawbridge to gate and cause massive pathfindinglag

### ISSUE THREAD

**Issue by Monsterfisch** [Sh0wdown/UnofficialCrusaderPatch#718]
Saladin in this case has workers stuck at the backgate with the drawbrdiges which causes the game to lag extensively (i have encountered this issue with other ais as well (usually with drawbridges in the back with a diagonal pathway and or stairs leading up to the drawbrdige, at least in the cases I have encountered)
[drawbridge to gate lag.zip](https://github.com/Sh0wdown/UnofficialCrusaderPatch/files/5663141/drawbridge.to.gate.lag.zip)

**Comment by Krarilotus** [comment]
This could have to do also with the bug, where troops sometimes get stuck on terrain and half killed walls, where they detect it to be pathable, when it isn't actually pathable. I suspect this has to do with the difference in terrain being smooth after a 20 hight difference or lower occurs, while the pathability only happens when terrain is 15 tiles high difference between adjacent tiles or lower.

**Comment by LordHansCapon** [comment]
Wish this would be an HD and not an Extreme save. We have mapped a lot of the HD version, not so much of the Extreme.

**Comment by Heroesflorian** [comment]
To add to that: Recently, @Monsterfisch also encountered this when he had gates adjacent to stockpiles (even when the gates faced in a direction so that the entrance tiles did not touch the stockpile directly).

**Comment by Krarilotus** [comment]
Yeah, i believe this happens also one tile away or if walls are connected
to the stockpile and extend its 'walkable' tiles it might even go further.

It could be related to the bug, where workers try to pathfind to
unaccessable tiles next to the stockpile, then don't find a way and bug
out, standing aeound for eternety. Or it could be related to the bug, where
you can build multiple granarys or armorys without them being attached, if
you build pocket walls next to the stockpile and out on the feelds to place
them inside of.
